services/device-registry-base/src/main/java/org/eclipse/hono/service/management/credentials/CommonCredentials.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.management.credentials;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Helper methods for {@link CommonCredential}.
+ */
+public final class CommonCredentials {
```

Comment:
how about simply adding these static methods to the existing *CommonCredential* class?

Reply:
I was following the pattern of splitting actual objects and helper classes apart, like `java.util.Collections`, `java.lang.Objects`, `com.google.collect.Streams`, …

Reply:
I see